x/claim/types/mission_test.go

```diff
@@ -0,0 +1,50 @@
+package types_test
+
+import (
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/stretchr/testify/require"
+
+	"github.com/tendermint/spn/testutil/sample"
+	claim "github.com/tendermint/spn/x/claim/types"
+)
+
+func TestMission_Validate(t *testing.T) {
+	invalidWeightGreaterThanOne := sample.Mission(r)
+	amt, err := sdk.NewDecFromStr("1.0000001")
+	require.NoError(t, err)
+	invalidWeightGreaterThanOne.Weight = amt
+
+	invalidWeightLessThanZero := sample.Mission(r)
+	amt, err = sdk.NewDecFromStr("-0.0000001")
+	require.NoError(t, err)
+	invalidWeightLessThanZero.Weight = amt
+
+	for _, tc := range []struct {
+		desc    string
+		mission claim.Mission
+		valid   bool
+	}{
+		{
+			desc:    "valid claim record",
+			mission: sample.Mission(r),
+			valid:   true,
+		},
+		{
+			desc:    "invalid weight - greater than 1",
+			mission: invalidWeightGreaterThanOne,
+			valid:   false,
+		},
+		{
+			desc:    "invalid weight - less than 0",
+			mission: invalidWeightLessThanZero,
+			valid:   false,
+		},
+	} {
+		tc := tc
```

Comment:
```suggestion
```

Why? `tc` is not a pointer

Reply:
This is actually a line we have in many of our tests.  I'm not sure tbh

Reply:
https://github.com/tendermint/spn/issues/806